File: codewar/Day-468/narcissistic.js

function narcissistic(value) {
  let res = [];
  let arrayOfNumber = [...String(value)].map(Number);
  for (let i = 0; i < arrayOfNumber.length; ++i){
    res.push(Math.pow(arrayOfNumber[i], arrayOfNumber.length));
  }
  return res.reduce((a, b) => a + b) === value ? true : false;
}

let a = 153;
console.log(narcissistic(a))

File: codewar/V1/Day-468/narcissistic.py
# https://www.codewars.com/kata/5287e858c6b5a9678200083c/train/python


def narcissistic( value ):
    res = []
    array_of_number = [int(n) for n in str(value)]
    for number in array_of_number:
        res.append(pow(number, len(array_of_number)))
    return sum(res) == value

print(narcissistic(153))